blocks/ifus/ifus.js

```diff
@@ -0,0 +1,248 @@
+import { translate } from '../../scripts/lib-franklin.js';
+
+function createDomStructure(structure, parentElement = document.body) {
+  structure.forEach((element) => {
+    const domElement = document.createElement(element.type);
+    if (element.attributes) {
+      Object.keys(element.attributes).forEach((attr) => {
+        domElement.setAttribute(attr, element.attributes[attr]);
+      });
+    }
+
+    if (element.textContent) {
+      domElement.textContent = element.textContent;
+    }
+
+    if (element.children) {
+      createDomStructure(element.children, domElement);
+    }
+
+    if (element.classes) {
+      element.classes.forEach((c) => domElement.classList.add(c));
+    }
+
+    if (element.position === 'prepend') {
+      parentElement.prepend(domElement);
+    } else {
+      parentElement.appendChild(domElement);
+    }
+  });
+}
+
+function populateSearch(
+  idSelector,
+  countrySelector,
+  searchButton,
+  result,
+  idProducer,
+  countryProducer,
+  assetFilter,
+  productCodes,
+  clear,
+) {
+  idProducer().forEach((value) => {
```

Comment:
option generation could be put into a generic function in order to avoid code duplication:

```
  const createOptions = (items, selector, attributes = {}) => {
    items.forEach((item) => {
      const option = document.createElement('option');
      for (const [key, value] of Object.entries(attributes(item))) {
        option.setAttribute(key, value);
      }
      option.textContent = item;
      selector.appendChild(option);
    });
  };

  createOptions(idProducer(), idSelector);
  idSelector.disabled = false;

  createOptions(
    countryProducer(),
    countrySelector,
    (country) => ({ value: country.ISO_3166_1_alpha_2_code, textContent: country['Country Name'] }),
  );
  countrySelector.disabled = false;
```

Reply:
I modified your suggestion slightly but adopted the spirit. Thanks!